### ISSUE TITLE
[TreeView] Using checkboxes and searching together causes an error

### ISSUE THREAD

**Issue by dtopalov** [telerik/kendo-angular#2171]
When Checkboxes are added to our [TreeView filtering example](https://www.telerik.com/kendo-angular-ui/components/treeview/how-to/filtering/), and the data is filtered so that the first root-level item disappears, and then the filter is removed so that it reappears, checking any of its children causes a JavaScript error:

[https://stackblitz.com/edit/angular-wgdqmx-nkfbtq?file=app/app.component.ts](https://stackblitz.com/edit/angular-wgdqmx-nkfbtq?file=app/app.component.ts)

Filter by "dec" - only Decor and its children remain.

Remove the filter - all items are visible.

Try to check some of Furniture's children.

P.S. Passing a copy of the item instead seems to resolve the problem:

    public search(items: any[], term: string): any[] {
        return items.reduce((acc, item) => {
            if (this.contains(item.text, term)) {
                acc.push(JSON.parse(JSON.stringify(item)));
            } else if (item.items && item.items.length > 0) {
                 const newItems = this.search(item.items, term);
                 if (newItems.length > 0) {
                acc.push({ text: item.text, items: newItems });
           }
        } 
        return acc;
       }, []);
    }`

**Comment by tsvetomir** [comment]
It seems to be sufficient to make a shallow copy of the item:

```ts
if (this.contains(item.text, term)) {
  const clone = Object.assign({}, item);
  acc.push(clone);
}
```

**Comment by Raisolution** [comment]
Example is updated accordingly.

**Comment by tsvetomir** [comment]
Fixed in the latest [development builds](https://www.telerik.com/kendo-angular-ui/components/installation/development-builds/).

**Comment by tsvetomir** [comment]
Released in v3.1.1
